cEP-0028.md

```diff
@@ -0,0 +1,150 @@
+   or hooking a section for a particular request type. For example:
+
+   ```
+    [longrunning]
+    enabled = True
+    cls_enabled = False
+    bears = SomeLongRunningBear
+   ```
+
+   This kind of a configuration will disable the bear completely during
+   analysis cycles of coala via a language server while enabling it for
+   regular invocations.
+
+   ```
+    [longrunning]
+    enabled = False
+    cls_on = save
+    files = *.py
+    bears = SomeLongRunningBear
+   ```
+
+   This example shows a configuration where the section is disabled for coala
+   but will only be enabled when the code editor dispatches a request of type
+   `textDocuemnt/didSave`.
+
+   ### Supported Parameters
+
+   The following are the supported parameter types:
+
+   - `cls_enabled`: A boolean indicating if the section should be enabled.
+     Disabling the section will turn if off for coala lang server. It will
+     default to true.
+
+   - `cls_on`: A list containing some or all of the available options,
+     `every`, `open`, `save`, `change`, `formatting`. This indicates the
+     type of request(s) the section should be executed on. `cls_on` will
+     default to `all`.
+
+   - `cls_fixes_enabled`: A boolean value indicating if the code fixes
+     generated by the section should be sent to the editor for being applied.
+     Initially only `formatting` request will respond with code fixes.
+
+   - `cls_as_global`: A boolean flag indicating if the section should be
+     considered as a global section. A global section executes analysis on
+     all of the files in its `files` attribute and reports diagnostic messages
+     on all the corresponding files open in the editor. If `cls_as_global` is
+     `false` then analysis runs only on the file currently being edited.
+     `cls_as_global` will default to false.
+
+2. Custom entry point into coala core
+   Since coala core was not designed to be a long running process coalib
+   does not provide a flexible entry point into it. To maximize the efficiency
+   of the coala language server, it will require an entry point that supports:
+
+   - Support for `FileProxies`: Since not all changes on the editor are flushed
+     to the disk coala core will have to support processing file content
+     directly as opposed to files.
+
+   - Support for preprocessing coala configurations: The configurations for
+     coala analysis need to be generated at runtime for each analysis cycle
+     based upon the `cls` parameters.
+
+   The following functions are core to the current general cli entry point and
+   need to be patched as part of the custom coala language server entry point
+   for:
+
+   - `coalib.processes.Processing.get_file_dict()`
+   - `coalib.processes.Processing.execute_section()`
+   - `coalib.coala_main.run_coala()`
+
+   The prototypes for functions that will replace/patch those are:
+
+   ```python
+    def get_file_dict(filename_list,
```

Comment:
If this is in coalib, and coala-ls is providing the file data, this 'get' is confusing.  'create_
..' would be clearer.

And why strange indent here?